File: 9.palindrome-number.py
#
# @lc app=leetcode id=9 lang=python3
#
# [9] Palindrome Number
#

# @lc code=start
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        else:
            x = str(x)
            l = len(x)
            for i in range(l // 2):
                if x[i] == x[l - 1 - i]:
                    continue
                else:
                    return False
            return True
# @lc code=end

File: 9.palindrome-number.java
/*
 * @lc app=leetcode id=9 lang=java
 *
 * [9] Palindrome Number
 */

// @lc code=start
class Solution {
    public boolean isPalindrome(int x) {
        long k = Math.toIntExact((long) x);
        if(x<0){
            return false;
        }
        long r = 0;
        while (x != 0) {
            r = r * 10 + x % 10;
            x /= 10;
        }
        return r==k;
    }
}
// @lc code=end

